### ISSUE TITLE
No-decoration mode

### ISSUE THREAD

**Issue by colinleroy** [gabe565/ascii-movie#35]
Hi,
Thanks for making this live :)
This does need a fullscreen, no-decoration mode though, for old monochrome terminals that don't like UTF-8 and color codes.

**Comment by gabe565** [comment]
Hey! Thanks for opening this. I was pretty sure that the tool I used would detect if a terminal supported colors. I guess my terminals are too fancy...I need to figure out how to reproduce.

**Comment by colinleroy** [comment]
To be more precise, there's no TERM set, it's over telnet. You would have to implement (a subset of) https://www.rfc-editor.org/rfc/rfc1091.html  (see also https://www.omnisecu.com/tcpip/telnet-negotiation.php)
basically you have to send 
IAC DO TERMINAL-TYPE

hope to receive
IAC WILL TERMINAL-TYPE

In which case you can open subnegotiation and send
IAC SB TERMINAL-TYPE SEND IAC SE

and the client will reply 
IAC SB TERMINAL_TYPE IS VT100 IAC SBE

as octal strings, this looks like:
IAC DO TERMINAL-TYPE = "\377\373\30"
IAC WILL TERMINAL-TYPE = "\377\373\30"
etc.
You can find the list of codes at https://www.ibm.com/docs/en/zos/2.1.0?topic=problems-telnet-commands-options

You can also negotiate the window size with the same system, basically.

(I'd love to have opened a PR but I do suck at Go and I'm already in another rabbit hole at the moment, learning 6502 assembly :laughing:)

**Comment by colinleroy** [comment]
If you want tests, I do have a vt100 implementation working well enough to handle vim and mc on my Apple II, so it should do the right thing :)

**Comment by gabe565** [comment]
Oh ok that makes total sense. Terminal detection works pretty well locally or over SSH, but I wrote the Telnet implementation from scratch, so it's not perfect yet. It already performs some detection so Telnet escape codes don't get sent if someone uses something like `nc` instead of the actual `telnet` command, so I'll see if I can hook into that. I'll let you know when I have something ready to test!

Also, completely off topic, but I recently tried my hand at writing an NES emulator in Go and was exposed to 6502 assembly more than I had been in the past...that seems like a pretty tough rabbit hole 😂

**Comment by gabe565** [comment]
@colinleroy Non-color terms are now supported! When a device connects, ascii-movie attempts to grab the current `TERM`, and depending on the value it will choose ANSI256, ANSI, or ASCII colors. This works for both telnet and SSH. If it is unable to fetch the current `TERM`, it will default to ANSI.

The overall look has some tweaks so the buttons still make sense. Here's what non-color mode looks like:
<img width="637" alt="image" src="https://github.com/gabe565/ascii-movie/assets/7717888/31d62770-a8c5-40fb-832d-069ebb26e4cf">

Would you mind testing on your old terminal? There may still be some UTF-8 characters I need to remove, but we'll see.

**Comment by colinleroy** [comment]
Hi Gabe! Thanks! I'll test that as soon as possible, should be in a few days 🙂

**Comment by colinleroy** [comment]
Hi Gabe! I've tested and it gives this :) https://www.colino.net/tmp/vt100.mp4
I think it will really need a no-decoration mode for vt100. Maybe I should implement this to inform the server of the window size (80x24 in my case) :https://datatracker.ietf.org/doc/html/rfc1073#section-2

Edit by @gabe565: Uploading the video here since the link is dead

https://github.com/user-attachments/assets/6113fca8-b883-402e-a1f1-ab6d760243be

**Comment by gabe565** [comment]
@colinleroy I love the video, but that's not as nice as I had hoped...I'll have to look at the vt100 spec some more!

**Comment by joeskeen** [comment]
+1 for this. It's cool that you have controls and all that, but it would be great if you could pass --no-controls or something and just have the video, with no borders, just plain.
